code/controllers/subsystem/mapping.dm

```diff
@@ -250,6 +250,15 @@ SUBSYSTEM_DEF(mapping)
 		if(islist(data["namelists"]))
 			S.name_categories = data["namelists"]
 
+		S.allowed_species = list()
+		if(islist(data["allowed_species"]))
+			var/list/allowed_species_list = data["allowed_species"]
+			for(var/species in allowed_species_list)
+				if(ispath(text2path(species), /datum/species) && species != "/datum/species")
+					S.allowed_species.Add(text2path(species))
```

Comment:
```suggestion
			for(var/species in allowed_species_list)
				if(species in GLOB.species_list)
					S.allowed_species.Add(GLOB.species_list[species])
```
Use the string IDs which are about to be macro'd in the KapuLimbs port instead of typepath strings.